docs/rfc/active/000-change-processor/README.md

```diff
@@ -0,0 +1,111 @@
+- Start Date: 2021-11-04
+- RFC PR:
+- Implementation PR(s):
+
+# Change processing
+
+## Summary
+
+DataHub is designed to collate externally managed metadata, as such it doesn't impose any rules on metadata changes
+(besides minimum data requirements to create entities/aspects). To start using DataHub for orchestration purposes we
+require the ability to flexibly validate/modify a change before it is applied.
+
+## Motivation
+
+We're looking to replace an in-house Data Catalog with DataHub. The catalog we're replacing has the ability to
+reject/ignore metadata changes based on a set of validation rules. The expected outcome would be that we would be able
+to specify rules that different entities/aspects would be required to follow such that we aren't required to fork
+DataHub.
+
+## Requirements
+
+- Validate metadata changes before they are applied based on properties of the incoming aspect change proposal. We
+  require the ability to ignore certain changes based on the previous value.
+- Easily create/change validation surrounding a metadata change.
+- Not maintain a fork of DataHub. We could replicate our desired behaviour by just modifying DataHub in a fork, however
+  one driver of replacing an internal catalog with DataHub is such that we no longer are required to maintain a service
+  that handles orchestration logic.
+
+## Detailed design
+
+The metadata-io package currently has a
+function ([updateLambda](https://github.com/linkedin/datahub/blob/352a0abf8d7e4dd5d5664a8c7cdf3d77bf6f1c51/metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java#L236))
+that is run before each aspect change proposal. It accepts the previous version of an aspect (or null if it's the first
+version) and the proposed aspect change. It currently does nothing with this information and returns the proposed new
+aspect value.
+
+We could extend this behaviour such that users can choose to overload it.
+
+### Proposed implementation
+
+1. On metadata-service start up, use a ServiceLoader to load a jar from a specified location (defined in config). In the
```

Comment:
Nice - whatever convention we use here (e.g. the path where plugin jars are added) we can reuse in other places as well. Actually I think the Entity Registry loader already does something quite similar to this